service/router.go

```diff
@@ -47,23 +48,41 @@ func SigningRouter() *mux.Router {
 	// Start the web service router
 	router := mux.NewRouter()
 
-	// API routes
 	router.Handle("/v1/version", Middleware(http.HandlerFunc(core.Version))).Methods("GET")
 	router.Handle("/v1/health", Middleware(http.HandlerFunc(core.Health))).Methods("GET")
-	router.Handle("/v1/serial", Middleware(ErrorHandler(sign.Serial))).Methods("POST")
-	router.Handle("/v1/request-id", Middleware(ErrorHandler(sign.RequestID))).Methods("POST")
-	router.Handle("/v1/model", Middleware(ErrorHandler(assertion.ModelAssertion))).Methods("POST")
-	router.Handle("/v1/pivot", Middleware(ErrorHandler(pivot.Model))).Methods("POST")
-	router.Handle("/v1/pivotmodel", Middleware(ErrorHandler(pivot.ModelAssertion))).Methods("POST")
-	router.Handle("/v1/pivotserial", Middleware(ErrorHandler(pivot.SerialAssertion))).Methods("POST")
-	router.Handle("/v1/pivotuser", Middleware(ErrorHandler(pivot.SystemUserAssertion))).Methods("POST")
+
+	// API routes
+	router.Handle("/v1/serial", metric.CollectAPIStats("signSerial",
+		Middleware(ErrorHandler(sign.Serial)))).
+		Methods("POST")
+	router.Handle("/v1/request-id", metric.CollectAPIStats("signRequestID",
+		Middleware(ErrorHandler(sign.RequestID)))).
+		Methods("POST")
+	router.Handle("/v1/model", metric.CollectAPIStats("assertionModelAssertion",
+		Middleware(ErrorHandler(assertion.ModelAssertion)))).
+		Methods("POST")
+	router.Handle("/v1/pivot", metric.CollectAPIStats("pivotModel",
+		Middleware(ErrorHandler(pivot.Model)))).
+		Methods("POST")
+	router.Handle("/v1/pivotmodel", metric.CollectAPIStats("pivotModelAssertion",
+		Middleware(ErrorHandler(pivot.ModelAssertion)))).
+		Methods("POST")
+	router.Handle("/v1/pivotserial", metric.CollectAPIStats("pivotSerialAssertion",
+		Middleware(ErrorHandler(pivot.SerialAssertion)))).
+		Methods("POST")
+	router.Handle("/v1/pivotuser", metric.CollectAPIStats("pivotSystemUserAssertion",
+		Middleware(ErrorHandler(pivot.SystemUserAssertion)))).
+		Methods("POST")
 
 	// Test log upload routes (only in the factory)
 	if datastore.InFactory() {
 		router.Handle("/testlog", Middleware(http.HandlerFunc(testlog.Index))).Methods("GET")
 		router.Handle("/testlog", Middleware(http.HandlerFunc(testlog.Submit))).Methods("POST")
 	}
 
+	// prometheus metrics endpint
+	router.Handle("/_status/metrics", metric.NewServer()).Methods("GET")
```

Comment:
Serial Vault runs as two different services, once for API and once for the Admin/UI. This is why we need to have this endpoint twice